src/gui_common/AddWindowReorderingSupportToSiblings.cs

```diff
@@ -0,0 +1,328 @@
+﻿using System.Collections.Generic;
+using Godot;
+using Godot.Collections;
+using static CustomWindow;
+
+/// <summary>
+///   Reorders windows by reordering their ancestors that are on the same level as this node.
+/// </summary>
+/// <remarks>
+///   <para>
+///     The windows are the ones who ask to establish the connection. If the window asks for a recursive
+///     connection then the connection gets passed to window reordering nodes in ancestors too.
+///   </para>
+/// </remarks>
+public class AddWindowReorderingSupportToSiblings : Control
+{
+    /// <summary>
+    ///   Paths to window reordering nodes in ancestors.
+    /// </summary>
+    [Export]
+    public Array<NodePath> WindowReorderingPaths = new();
+
+    /// <summary>
+    ///   Finds first window reordering node in ancestors to connect to.
+    /// </summary>
+    /// <remarks>
+    ///   <para>
+    ///     Ignored when window reordering paths are not empty.
+    ///   </para>
+    /// </remarks>
+    [Export]
+    public int AutomaticWindowReorderingDepth = 5;
+
+#pragma warning disable CA2213
+
+    /// <summary>
+    ///   Pairs containing window reordering nodes and what sibling of theirs is an ancestor of this class.
+    /// </summary>
+    /// <remarks>
+    ///   <para>
+    ///     Used to pass the connections from windows.
+    ///   </para>
+    /// </remarks>
+    private List<(AddWindowReorderingSupportToSiblings Node, Node Sibling)> windowReorderingAncestors = new();
+
+    /// <summary>
+    ///   A sibling that is an ancestor of a window that is currently on top of others.
```

Comment:
I think I still don't fully understand this variable. Could you add a short remarks section detailing an example or two of in which situations this variable is used?

Reply:
Sure. I have written something that should make it clear I hope. In short it's a variable used to check if the window is already on the top and if not then its index is used to move current window on the top (and then the moved windows Ancestor becomes new topSibling).
It doesn't have to be saved, but I don't think that it hurts to save it and just update it after changes.